govuk_chat_evaluation/rag_answers/custom_deepeval/metrics/coherence/metric.py

```diff
@@ -0,0 +1,193 @@
+
+        self.model, self.using_native_model = initialize_model(model)
+        self.evaluation_model = self.model.get_model_name()
+        self.evaluation_cost: float | None = None
+
+        self.score: float | None = None
+        self.reason: str | None = None
+        self.raw_score: int | None = None
+
+    async def a_measure(
+        self,
+        test_case: LLMTestCase,
+        _show_indicator: bool = True,
+        _in_component: bool = False,
+    ) -> float:
+        check_llm_test_case_params(test_case, self._required_params, self)
+
+        if self.using_native_model:
+            self.evaluation_cost = 0.0
+        else:
+            self.evaluation_cost = None
+
+        with metric_progress_indicator(
+            self,
+            async_mode=True,
+            _show_indicator=_show_indicator,
+            _in_component=_in_component,
+        ):
+            self.error = None
+            prompt = self.evaluation_template.evaluate(
+                user_input=test_case.input or "",
+                actual_output=test_case.actual_output or "",
+                evaluation_steps=self.evaluation_steps,
+                rubric=self.rubric,
+                score_range=self.score_range,
+                additional_context=self.additional_context,
+            )
+
+            try:
+                judgement = await self._generate_result_from_model(
+                    prompt, schema=CoherenceJudgement
+                )
+            except Exception as exc:
+                self.error = f"Failed to generate coherence judgement: {exc}"
+                self.score = float("nan")
+                self.success = False
+                self.reason = None
+                logging.error(self.error, exc_info=exc)
+                return float("nan")
+
+            raw_score = self._coerce_and_clamp_score(judgement.score)
+            self.raw_score = raw_score
+            self.score = self._normalize(raw_score, self.score_range)
+            cleaned_reason = judgement.reason.strip()
+            self.reason = cleaned_reason if self.include_reason else None
+            self.success = self.is_successful()
+
+            capture_metric_type(
+                self.__name__, async_mode=True, in_component=_in_component
+            )
+
+            verbose_reason = cleaned_reason if self.include_reason else "Reason omitted"
+            self.verbose_logs = construct_verbose_logs(
+                self,
+                steps=[
+                    f"Raw Score: {raw_score}",
+                    f"Score: {self.score}",
+                    f"Reason: {verbose_reason}",
+                ],
+            )
+
+            return self.score
+
+    async def _generate_result_from_model(
+        self, prompt: str, schema: Type[CoherenceJudgement]
+    ) -> CoherenceJudgement:
+        if self.using_native_model:
+            raw_result, raw_cost = await self.model.a_generate(prompt, schema=schema)
```

Comment:
Prefixing these with raw seems a bit unnecessary as we don't have a non raw alternative.